libsawtooth/src/receipt/store/diesel/operations/add_txn_receipts.rs

```diff
@@ -47,6 +49,133 @@ pub(in crate::receipt::store::diesel) trait ReceiptStoreAddTxnReceiptsOperation
     fn add_txn_receipts(&self, receipts: Vec<TransactionReceipt>) -> Result<(), ReceiptStoreError>;
 }
 
+#[cfg(feature = "postgres")]
+impl<'a> ReceiptStoreAddTxnReceiptsOperation
+    for ReceiptStoreOperations<'a, diesel::pg::PgConnection>
+{
+    fn add_txn_receipts(&self, receipts: Vec<TransactionReceipt>) -> Result<(), ReceiptStoreError> {
+        self.conn.transaction::<(), _, _>(|| {
+            let index: i64 = match transaction_receipt::table
+                .order(transaction_receipt::idx.desc())
+                .first::<TransactionReceiptModel>(self.conn)
+                .optional()?
+            {
+                Some(rec) => rec.idx,
+                None => 0,
+            };
+            for (i, receipt) in (1..).zip(receipts.into_iter()) {
+                let id = &receipt.transaction_id;
+
+                // Create the TransactionReceiptModel and insert it into the
+                // transaction_receipt table
+                let transaction_receipt_model = TransactionReceiptModel {
+                    transaction_id: id.to_string(),
+                    idx: index + i,
+                };
+                insert_into(transaction_receipt::table)
+                    .values(transaction_receipt_model)
+                    .execute(self.conn)?;
+
+                match &receipt.transaction_result {
+                    TransactionResult::Valid{ state_changes, events, data } => {
+                        // Create a vector of `ValidTransactionResultDataModels` from
+                        // the valid transaction result's data field
+                        let valid_transaction_result_data_models: Vec<NewValidTransactionResultDataModel> = data
+                            .iter()
+                            .enumerate()
+                            .map(|(i, d)| {
+                                let index = match i32::try_from(i) {
+                                    Ok(index) => index,
+                                    Err(_) => return Err(ReceiptStoreError::InternalError(InternalError::with_message("Unable to convert index into i32".to_string()))),
```

Comment:
Please properly wrap this line at 100 characters

Reply:
Fixed